daemon/cmd/endpoint.go

```diff
@@ -533,14 +534,18 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) (resp middleware.Resp
 	} else {
 		log.WithField(logfields.Params, logfields.Repr(params)).Debug("PUT /endpoint/{id} request")
 	}
-	epTemplate := params.Endpoint
+
+	if err := server.CheckAPIMutationPermissions(params); err != nil {
```

Comment:
https://goswagger.io/use/middleware.html

Reply:
I had a bit of an out-of-band discussion with @joamaki around this, and we think there's an option here. I had previously avoided this because I didn't want to get too deep into URL parsing in order to determine which APIs should be made available or not. It should be sufficient to categorize the APIs into groups like `endpoint`, `ipam`, `policy` etc. and expose those groups. This will make it easier to integrate as middleware rather than sprinkling authorization checks throughout all of the handlers.